src/realm/object-store/results.cpp

```diff
@@ -350,12 +361,20 @@ Mixed Results::get_any(size_t ndx)
         case Mode::Collection:
             evaluate_sort_and_distinct_on_collection();
             if (m_list_indices) {
-                if (ndx < m_list_indices->size())
-                    return m_collection->get_any((*m_list_indices)[ndx]);
+                ndx = (ndx < m_list_indices->size()) ? (*m_list_indices)[ndx] : realm::npos;
             }
-            else {
-                if (ndx < m_collection->size())
-                    return m_collection->get_any(ndx);
+            if (ndx < m_collection->size()) {
+                Mixed mixed;
+                if (m_dictionary_keys) {
+                    if (auto dict = dynamic_cast<realm::Dictionary*>(m_collection.get())) {
```

Comment:
It opens up an edge case that if SDKs construct a Results with `as_keys == true` for a list or set, all elements will be null. I would almost say that case is eligible for an assertion because it is API misuse. Even better would be to make it impossible for SDKs to construct this combination, but I can't see any easy way to do that. Maybe a Results of all nulls is also acceptable, up to you.

Reply:
@ironage I made an alternative implementation in an additional commit. Perhaps it reduces the chance of mistakes. What do you think?

Reply:
nice, I am much more comfortable with this 👍